src/States/Elytra.cs

```diff
@@ -431,19 +431,26 @@ private static PlayerDeadBody Mod_Player_Die(On.Celeste.Player.orig_Die orig, Pl
             Audio.Stop(eventInstance);
 
         return orig(self, direction, evenIfInvincible, registerDeathInStats);
-    }
-
-    private static void Mod_PlayerSprite_ctor(On.Celeste.PlayerSprite.orig_ctor orig, PlayerSprite self, PlayerSpriteMode mode)
-    {
-        orig(self, mode);
-
-        self.Animations[ELYTRA_ANIM] = new()
-        {
-            Frames = GFX.Game.GetAtlasSubtextures("characters/player_no_backpack/CommunalHelper/fly").ToArray(),
-            Delay = 10f,
-        };
-    }   
-
+    }
+
+    private static void Mod_PlayerSprite_CreateFramesMetadata(On.Celeste.PlayerSprite.orig_CreateFramesMetadata orig, string sprite)
+    {
+        SpriteData data = GFX.SpriteBank.SpriteData[sprite];
+        if (!data.Sprite.Has(ELYTRA_ANIM))
+        {
+            string path = !string.IsNullOrEmpty(data.Sources[0].OverridePath) ? data.Sources[0].OverridePath : data.Sources[0].Path;
+            if (GFX.Game.HasAtlasSubtextures($"{path}CommunalHelper/fly"))
+                path = "characters/player_no_backpack/";

```

Comment:
```suggestion
            if (!GFX.Game.HasAtlasSubtextures($"{path}CommunalHelper/fly"))
                path = "characters/player_no_backpack/";
```
Feel like you wanted to fallback to the `player_no_backpack`, but right now you always use that sprite.

Reply:
> Feel like you wanted to fallback to the `player_no_backpack`, but right now you always use that sprite.

Ah sorry for that, done now